lib/worker-type.js

```diff
@@ -794,7 +794,7 @@ WorkerType.prototype.determineSpotBids = function (managedRegions, pricing, chan
     } else {
       debug('WorkerType %s is exceeding its max price of %d with %d',
             this.workerType, this.maxPrice, cheapestPrice);
-      return [];
+      return spotBids;
```

Comment:
Why is this inside a loop... I'm not sure... ie. how can `cheapestPrice` be different in each iteration of the `while (change > 0) {` loop???

I might be missing something... But why don't we refactor into some thing sane...

1) determine list of bids for various machine/region/zone/capacity combinations
2) `while(change > 0) {` pick random bids to make from 3 cheapest options below `maxPrice`, if there is nothing below `maxPrice` pick randomly between all options give `price = maxPrice` (so we'll get nodes when capacity is available).

Not sure about the last part... But definitely simplify this code...